src/metabase/models/pulse.clj

```diff
@@ -222,6 +214,11 @@
   [notification-or-id]
   (t2/select PulseChannel, :pulse_id (u/the-id notification-or-id)))
 
+(def ^:dynamic *allow-hydrate-archived-cards*
```

Comment:
It's a shame we need to use dynamic vars instead of passing an arg to a `t2/hydrate` call. This is okay for now but we should look into extending toucan's `hydrate` function asap so we don't need to keep doing this. This is one of the biggest reasons why we can't get rid of dynamic vars.